AppCenter/AppCenter/Internals/MSAppCenterPrivate.h

```diff
@@ -1,11 +1,13 @@
 #import "MSAppCenter.h"
 #import "MSChannelUnitProtocol.h"
-#import "MSOneCollectorChannelDelegate.h"
 
-@interface MSAppCenter ()
+@class MSAppCenterIngestion;
+@class MSOneCollectorChannelDelegate;
 
-@property(nonatomic) id <MSChannelUnitProtocol> channelUnit;
+@interface MSAppCenter ()
 
+@property(nonatomic) MSAppCenterIngestion *ingestion;
```

Comment:
I'm not sure this is a good idea to be placed in AppCenter class. @ElektrojungeMS @clpolet What do you guys think?

Reply:
It is already stored in the channel group, but we need to access to it from `MSAppCenter` class to update the app secret. Storing the pointer is better than get it for channel group and blindly cast protocol to implementation.

Reply:
Any reason to have a reference to Ingestion from AppCenter class in terms of design? I don't see any reasons, "better" is just for convenience.

Reply:
The other possibility is to forward the AppSecret to the channelGroup which will in turn set the AppSecret to its ingestion. Then no ingestion in AppCenter class.

Reply:
@jaelim-ms and I are talking about this right now, too. We both think that it's wrong to have a reference to ingestion in `MSAppCenter`. We favor what @clpolet is suggesting, how about we add an API to `MSChannelGroupProtocol` to update the app secret. We can use `setAppSecret:(NSString *)appSecret` for now and if we need a more generic name in the future, we can rename the api as it's not public.

Reply:
This will also align the design with Android. ChannelGroup is the one that keeps all information and instances, coordinate core service in it. ChannelGroup is the right place to keep app secret, it can receive an app secret as @ElektrojungeMS mentioned and then set it to Ingestion would be the best approach.

Reply:
Thanks guys, I agree, it's better. 👍